pkg/chartmuseum/server/multitenant/routes.go

```diff
@@ -4,21 +4,34 @@ import (
 	cm_router "github.com/kubernetes-helm/chartmuseum/pkg/chartmuseum/router"
 )
 
-func (s *MultiTenantServer) Routes() []cm_router.Route {
-	var routes []cm_router.Route
+func (s *MultiTenantServer) Routes() []*cm_router.Route {
+	var routes []*cm_router.Route
 
-	serverInfoRoutes := []cm_router.Route{
-		{"READ", "GET", "/", s.defaultHandler},
-		{"SYSTEM", "GET", "/health", s.getHealthCheckHandler},
+	serverInfoRoutes := []*cm_router.Route{
+		{"GET", "/", s.getWelcomePageHandler, cm_router.RepoPullAction},
+		{"GET", "/health", s.getHealthCheckHandler, cm_router.SystemInfoAction},
 	}
 
-	helmChartRepositoryRoutes := []cm_router.Route{
-		{"READ", "GET", "/:repo/index.yaml", s.getIndexFileRequestHandler},
-		{"READ", "GET", "/:repo/charts/:filename", s.getStorageObjectRequestHandler},
+	helmChartRepositoryRoutes := []*cm_router.Route{
+		{"GET", "/:repo/index.yaml", s.getIndexFileRequestHandler, cm_router.RepoPullAction},
+		{"GET", "/:repo/charts/:filename", s.getStorageObjectRequestHandler, cm_router.RepoPullAction},
+	}
+
+	chartManipulationRoutes := []*cm_router.Route{
+		{"GET", "/api/:repo/charts", s.getAllChartsRequestHandler, cm_router.RepoPullAction},
```

Comment:
Maybe these constants should be RepoReadAction and RepoWriteAction?

Reply:
I think "pull" and "push" are actually more appropriate here, if we are trying to closely match Dockers JWT format:
```
{
    "iss": "auth.docker.com",
    "sub": "jlhawn",
    "aud": "registry.docker.com",
    "exp": 1415387315,
    "nbf": 1415387015,
    "iat": 1415387015,
    "jti": "tYJCO1c6cnyy7kAn0c7rKPgbV1H1bFws",
    "access": [
        {
            "type": "repository",
            "name": "samalba/my-app",
            "actions": [
                "pull",
                "push"
            ]
        }
    ]
}
```
https://docs.docker.com/registry/spec/auth/jwt/